src/ng/directive/input.js

```diff
@@ -1358,17 +1361,21 @@ function createDateInputType(type, regexp, parseDate, format) {
       });
     }
 
-    function isValidDate(value) {
-      // Invalid Date: getTime() returns NaN
-      return value && !(value.getTime && value.getTime() !== value.getTime());
-    }
-
     function parseObservedDateValue(val) {
-      return isDefined(val) && !isDate(val) ? parseDate(val) || undefined : val;
+      if (val === '') {
+        return undefined;
+      } else {
+        return isDefined(val) && !isDate(val) ? parseDate(val) || undefined : val;
+      }
     }
   };
 }
 
+function isValidDate(value) {
+  // Invalid Date: getTime() returns NaN
+  return value && !(value.getTime && value.getTime() !== value.getTime());
```

Comment:
Unrelated to this PR: 
That's a weird validation function (e.g. `isValidDate({})` -- true).
I would expect `value && value.getTime && value.getTime() === value.getTime()`.